sdk/eventgrid/azure-messaging-eventgrid-cloudnative-cloudevents/src/main/java/com/azure/messaging/eventgrid/cloudnative/cloudevents/EventGridCloudNativeEventPublisher.java

```diff
@@ -0,0 +1,176 @@
+        EventGridPublisherAsyncClient<com.azure.core.models.CloudEvent> asyncClient, Iterable<CloudEvent> events) {
+        return asyncClient.sendEvents(toEventGridCloudEvents(events));
+    }
+
+    /**
+     * Publishes the given native cloud events to the set topic or domain and gives the response issued by EventGrid.
+     *
+     * @param syncClient a service client that publishes events to an EventGrid topic or domain.
+     * @param events the native cloud events to publish.
+     * @param context the context to use along the pipeline.
+     */
+    public static void sendEventsWithResponse(EventGridPublisherClient<com.azure.core.models.CloudEvent> syncClient,
+        Iterable<CloudEvent> events, Context context) {
+        syncClient.sendEventsWithResponse(toEventGridCloudEvents(events), context);
+    }
+
+    /**
+     * Publishes the given native cloud events to the set topic or domain and gives the response issued by EventGrid.
+     *
+     * @param asyncClient a service asynchronous client that publishes events to an EventGrid topic or domain.
+     * @param events the native cloud events to publish.
+     * @return the response from the EventGrid service.
+     */
+    public static Mono<Response<Void>> sendEventsWithResponseAsync(
+        EventGridPublisherAsyncClient<com.azure.core.models.CloudEvent> asyncClient, Iterable<CloudEvent> events) {
+        return asyncClient.sendEventsWithResponse(toEventGridCloudEvents(events));
+    }
+
+    private static Iterable<com.azure.core.models.CloudEvent> toEventGridCloudEvents(Iterable<CloudEvent> events) {
+        List<com.azure.core.models.CloudEvent> cloudEvents = new ArrayList<>();
+        for (CloudEvent event : events) {
+            cloudEvents.add(toEventGridCloudEvent(event));
+        }
+        return cloudEvents;
+    }
+
+    private static com.azure.core.models.CloudEvent toEventGridCloudEvent(CloudEvent event) {
+        if (event == null) {
+            return null;
+        }
+
+        // Identify data format by data content type
+        CloudEventDataFormat dataFormat;
+        final String dataContentType = event.getDataContentType();
+        if (dataContentType == null) {
+            dataFormat = CloudEventDataFormat.JSON;
+        } else {
+            // Future version:
+            // https://github.com/cloudevents/spec/blob/main/cloudevents/formats/json-format.md#311-payload-serialization
+            // That is, a datacontenttype declares JSON-formatted content if its media type, when stripped of parameters,
+            // has the form */json or */*+json. If the datacontenttype is unspecified, processing SHOULD proceed as if
+            // the datacontenttype had been specified explicitly as application/json.
+
+            // Current version: https://github.com/cloudevents/sdk-java/blob/ff07dd83150deeb5eca5405cb80760b71e7b470c/
+            // formats/json-jackson/src/main/java/io/cloudevents/jackson/JsonFormat.java#L146-L149
```

Comment:
We should include the version number in the comment instead of using "Future version" and "Current version".